src/pages-and-resources/discussions/data/thunks.js

```diff
@@ -0,0 +1,25 @@
+import { getDiscussionApps } from './api';
+import { addModels } from '../../../generic/model-store';
+import {
+  FAILED, fetchAppsSuccess, LOADING, updateStatus,
+} from './slice';
+
+/* eslint-disable import/prefer-default-export */
+export function fetchApps(courseId) {
+  return async (dispatch) => {
+    dispatch(updateStatus({ courseId, status: LOADING }));
+
+    try {
+      const { apps, features } = await getDiscussionApps(courseId);
+
+      dispatch(addModels({ modelType: 'apps', models: apps }));
+      dispatch(addModels({ modelType: 'features', models: features }));
+      dispatch(fetchAppsSuccess({
+        appIds: apps.map(app => app.id),
+        featureIds: features.map(feature => feature.id),
+      }));
+    } catch (error) {
+      dispatch(updateStatus({ courseId, status: FAILED }));
```

Comment:
This is rudimentary error handling and doesn't get surfaced to the user yet.

Reply:
do we have that called out in a ticket?

Reply:
My plan was just to leave this in the TODO since this code is still all pretty likely to change/evolve as we work on polishing it.  It didn't feel like it rose to the level of a separate ticket.  (That's mostly just based on personal process...)

Reply:
k I'll quit asking... except for my question about an i18n ticket